x-pack/plugins/ml/public/application/timeseriesexplorer/components/plot_function_controls/plot_function_controls.tsx

```diff
@@ -30,11 +34,57 @@ const plotByFunctionOptions = [
 export const PlotByFunctionControls = ({
   functionDescription,
   setFunctionDescription,
+  selectedDetectorIndex,
+  selectedJobId,
+  selectedEntities,
 }: {
   functionDescription: undefined | string;
   setFunctionDescription: (func: string) => void;
+  selectedDetectorIndex: any;
+  selectedJobId: string;
+  selectedEntities: Record<string, any>;
 }) => {
+  const toastNotificationService = useToastNotificationService();
+
+  useEffect(() => {
+    if (functionDescription !== undefined) {
+      return;
+    }
+    const getFunctionDescriptionToPlot = async () => {
+      const functionToPlot = await getFunctionDescription(
+        {
+          selectedDetectorIndex,
+          selectedEntities,
+          selectedJobId,
+          selectedJob,
+        },
+        toastNotificationService
+      );
+      setFunctionDescription(functionToPlot);
+    };
```

Comment:
It would be better to init this function outside of the effect

Reply:
Updated here https://github.com/elastic/kibana/pull/83507/commits/1b10541a29c1a6c8552a9538bc506796a8110fd3